src/utils/containers.ts

```diff
@@ -196,13 +197,12 @@ export class SleepList<T, ID> {
     // timestamp to fire at, T
     list: [number, T][] = [];
     timer: NodeJS.Timeout | null = null;
-    handler: (item: T) => Promise<void>;
-    get_id: (item: T) => ID;
 
-    constructor(handler: (item: T) => Promise<void>, get_id: (item: T) => ID) {
-        this.handler = handler;
-        this.get_id = get_id;
-    }
+    constructor(
+        private wheatley: Wheatley,
```

Comment:
Having to pass Wheatley here definitely doesn't feel great... But note much can be done other than making the Wheatley instance global, which I'd briefly wondered about in the past.

Reply:
Yeah I was wondering about this one. I think the alternative would be to just do the error handling outside of the timer if that's possible?

Reply:
I'd leave it like this for now, it's only used in one place anyways. I think I'd rather avoid introducing a global Wheatley and making code depend on that…